PowerVCF.psm1

```diff
@@ -54,40 +54,101 @@ Function Request-VCFToken {
+
+        .EXAMPLE
+        $credential = Get-Credential
+        Request-VCFToken -Fqdn sfo-vcf01.sfo.rainpole.io -Credential $credential
+        This example shows how to connect to the SDDC Manager instance.
+
+        .EXAMPLE
+        Request-VCFToken -Fqdn sfo-vcf01.sfo.rainpole.io
+        This example shows how to connect to the SDDC Manager instance.
+        The operator will be prompted for a username and password.
+
+        .PARAMETER Fqdn
         The fully qualified domain name of the SDDC Manager instance.
 
-        .PARAMETER username
+        .PARAMETER UserName
         The username to authenticate to the SDDC Manager instance.
 
-        .PARAMETER password
-        The password to authenticate to the SDDC Manager instance.
+        .PARAMETER Password
+        The password to authenticate to the SDDC Manager instance. 
+        This parameter takes either a string or a SecureString variable.
+        If not specified, a SecureString variable will be prompted for.
+
+        .PARAMETER Credential
+        Specifies a user account to authenticate to the SDDC Manager instance.
 
         .PARAMETER skipCertificateCheck
         Switch to skip certificate check when connecting to the SDDC Manager instance.
     #>
 
+    [CmdletBinding(DefaultParameterSetName = 'PSCredentialSet')]
     Param (
-        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$fqdn,
-        [Parameter (Mandatory = $false)] [ValidateNotNullOrEmpty()] [String]$username,
-        [Parameter (Mandatory = $false)] [ValidateNotNullOrEmpty()] [String]$password,
-        [Parameter (Mandatory = $false)] [ValidateNotNullOrEmpty()] [Switch]$skipCertificateCheck
+        [Parameter (Mandatory = $true, ParameterSetName = 'UserNameAndPasswordSet')]
+    	[Parameter (Mandatory = $true, ParameterSetName = 'PSCredentialSet')]
+        [ValidateNotNullOrEmpty()]
+        [string]$Fqdn,
+
+        [Parameter (Mandatory = $true, ParameterSetName = 'UserNameAndPasswordSet')]
+      	[ValidateNotNullOrEmpty()]
+      	[string]$UserName,
+
+        [Parameter (Mandatory = $false, ParameterSetName = 'UserNameAndPasswordSet')]
+        [ValidateNotNullOrEmpty()]
+        [Object]$Password,
+
+        [Parameter (Mandatory = $true, ParameterSetName = 'PSCredentialSet')]
+        [System.Management.Automation.PSCredential]
+        [System.Management.Automation.Credential()]$Credential,            
+
+        [Parameter (Mandatory = $false)]
+        [ValidateNotNullOrEmpty()] 
+        [Switch]$SkipCertificateCheck
     )
 
-    if ( -not $PsBoundParameters.ContainsKey("username") -or ( -not $PsBoundParameters.ContainsKey("password"))) {
-        $creds = Get-Credential # Request Credentials
-        $username = $creds.UserName.ToString()
-        $password = $creds.GetNetworkCredential().password
+    try {
+        if ($PSCmdlet.ParameterSetName -eq 'UserNameAndPasswordSet') {
+            $user = $UserName
```

Comment:
What is the need to declare and use a new var?

Reply:
I figured that since ```$UserName``` is used as part of that particular parameter set (username + password combo), it'd be prudent to have one that can be used by both sets, with a different name than the parameter name/variable.